test/e2e/apps/deployment.go

```diff
@@ -134,6 +141,238 @@ var _ = SIGDescribe("Deployment", func() {
+				Replicas: &testDeploymentDefaultReplicas,
+				Selector: &testDeploymentLabelSelectors,
+				Template: v1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: testDeploymentLabelSelectors.MatchLabels,
+					},
+					Spec: v1.PodSpec{
+						Containers: []v1.Container{{
+							Name:  testDeploymentName,
+							Image: testDeploymentInitialImage,
+						}},
+					},
+				},
+			},
+		}
+		_, err := f.ClientSet.AppsV1().Deployments(testNamespaceName).Create(context.TODO(), &testDeployment, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "failed to create Deployment %v in namespace %v", testDeploymentName, testNamespaceName)
+
+		ginkgo.By("watching for the Deployment to be added")
+		dplmtWatchTimeoutSeconds := int64(180)
+		dplmtWatch, err := f.ClientSet.AppsV1().Deployments(testNamespaceName).Watch(context.TODO(), metav1.ListOptions{LabelSelector: testDeploymentLabelsFlat, TimeoutSeconds: &dplmtWatchTimeoutSeconds})
+		framework.ExpectNoError(err, "Failed to setup watch on newly created Deployment")
+
+		foundEvent := false
+		dplmtWatchChan := dplmtWatch.ResultChan()
+		for event := range dplmtWatchChan {
+			if event.Type == watch.Added {
+				foundEvent = true
+				break
+			}
+		}
+		framework.ExpectEqual(foundEvent, true, "failed to find watch event %v", watch.Added)
+
+		ginkgo.By("waiting for all Replicas to be Ready")
+		foundEvent = false
+		for event := range dplmtWatchChan {
+			deployment, ok := event.Object.(*appsv1.Deployment)
+			framework.ExpectEqual(ok, true, "unable to convert event.Object type")
+			if deployment.Status.AvailableReplicas == testDeploymentDefaultReplicas &&
+				deployment.Status.ReadyReplicas == testDeploymentDefaultReplicas {
+				foundEvent = true
+				break
+			}
+		}
+		framework.ExpectEqual(foundEvent, true, "failed to see scale of replicas")
+
+		ginkgo.By("patching the Deployment")
+		deploymentPatch, err := json.Marshal(map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"labels": map[string]string{"test-deployment": "patched"},
+			},
+			"spec": map[string]interface{}{
+				"replicas": testDeploymentMinimumReplicas,
+				"template": map[string]interface{}{
+					"spec": map[string]interface{}{
+						"containers": [1]map[string]interface{}{{
+							"name":    testDeploymentName,
+							"image":   testDeploymentPatchImage,
+							"command": []string{"/bin/sleep", "100000"},
+						}},
+					},
+				},
+			},
+		})
+		framework.ExpectNoError(err, "failed to Marshal Deployment JSON patch")
+		_, err = f.ClientSet.AppsV1().Deployments(testNamespaceName).Patch(context.TODO(), testDeploymentName, types.StrategicMergePatchType, []byte(deploymentPatch), metav1.PatchOptions{})
+		framework.ExpectNoError(err, "failed to patch Deployment")
+
+		foundEvent = false
+		for event := range dplmtWatchChan {
+			if event.Type == watch.Modified {
+				foundEvent = true
+				break
+			}
+		}
+		framework.ExpectEqual(foundEvent, true, "failed to see scale of replicas")
```

Comment:
This isn't waiting for scale, the next loop is.  I would remove this, it seems redundant.

Reply:
The comment has been fixed. I believe it's good to have a confirm on the patch/update when it comes through for data/resource consistency (such as, we have the correct _resourceVersion_, data, etc...)